presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloRecordCursor.java

```diff
@@ -0,0 +1,584 @@
+     *
+     * @return Total bytes
+     */
+    @Override
+    public long getTotalBytes()
+    {
+        return 0L; // unknown value
+    }
+
+    /**
+     * Gets the bytes that have been processed so far
+     *
+     * @return Completed bytes
+     */
+    @Override
+    public long getCompletedBytes()
+    {
+        return bytesRead;
+    }
+
+    /**
+     * Get the amount of time spent reading data, in nanoseconds
+     *
+     * @return Read time nanos
+     */
+    @Override
+    public long getReadTimeNanos()
+    {
+        return nanoStart > 0L ? (nanoEnd == 0 ? System.nanoTime() : nanoEnd) - nanoStart : 0L;
+    }
+
+    /**
+     * Gets the Presto type for the given field ordinal
+     *
+     * @param field Ordinal of the field aka column
+     * @return Presto type for this field
+     */
+    @Override
+    public Type getType(int field)
+    {
+        checkArgument(field < cHandles.size(), "Invalid field index");
+        return cHandles.get(field).getType();
+    }
+
+    /**
+     * Advances the iterator to the next Presto row
+     *
+     * @return True if a row was read, false otherwise
+     */
+    @Override
+    public boolean advanceNextPosition()
+    {
+        if (nanoStart == 0) {
+            nanoStart = System.nanoTime();
+        }
+
+        // In this method, we are effectively doing what the WholeRowIterator is doing w/o the extra
+        // overhead. We continually read key/value pairs from Accumulo, deserializing each entry
+        // using the instance of AccumuloRowSerializer we were given until we see the row ID change.
+
+        try {
+            // We start with the end! When we have no more key/value pairs to read
+
+            // If the iterator doesn't have any more values
+            if (!iterator.hasNext()) {
+                // Deserialize final KV pair
+                // This accounts for the edge case when the last read KV pair
+                // was a new row and we broke out of the below loop
+                if (prevKV != null) {
+                    serializer.reset();
+                    serializer.deserialize(prevKV);
+                    prevKV = null;
+                    return true;
+                }
+                else {
+                    // else we are super done
+                    return false;
+                }
+            }
+
+            // If we do have key/value pairs to process from Accumulo, we reset and deserialize the
+            // previously scanned key/value pair as we have started a new row.
+            // This code block does not execute the first time this method is called
+            if (prevRowID.getLength() != 0) {
+                serializer.reset();
+                serializer.deserialize(prevKV);
+            }
+
+            // While there are key/value pairs to read and we don't break out of the loop
+            boolean braek = false;
+            while (iterator.hasNext() && !braek) {
```

Comment:
Cannot you inline the condition from https://github.com/prestodb/presto/pull/5030/files#diff-d6a7c7acc93e774b949565cf9f619260R258 here instead of using `braek` variable?